### PR TITLE
James 1777 2

### PR DESCRIPTION
**Pull request by aduprat**
Rebased of #295, handle attachment name.

Taking care of last comment in the last fixup commit.

### CODE REVIEW

File: mailbox/store/src/test/resources/eml/oneAttachmentWithoutName.eml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Content-Disposition: inline
+
+<b>Content of part 1-2</b>
+------=_Part_2_1330682067197--
+
+------=_Part_0_1330682067197
+Content-Type: application/octet-stream;
+Content-Transfer-Encoding: base64
+Content-Disposition: attachment;
+	filename="gimp.png"
```
Comment (rouazana) [inline mailbox/store/src/test/resources/eml/oneAttachmentWithoutName.eml]: that's not a name?

Comment (aduprat) [inline mailbox/store/src/test/resources/eml/oneAttachmentWithoutName.eml]: No, the name is taken from the Content-Type header.

Comment (rouazana) [inline mailbox/store/src/test/resources/eml/oneAttachmentWithoutName.eml]: ok :)

**Comment by rouazana** [comment]
:+1:

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature [hunk 2]
```diff
@@ -15,3 +15,9 @@ Feature: Download GET
   Scenario: Getting an attachment with an unknown blobId
     When "username@domain.tld" downloads "123"
     Then the user should receive a not found response
+
+  Scenario: Getting an attachment previously stored with a desired name
+    Given "username@domain.tld" mailbox "inbox" contains a message "1" with an attachment "2"
+    When "username@domain.tld" downloads "2" with "myFileName.txt" name
+    Then the user should receive that attachment
+    And the response contains a Content-Disposition header file with "myFileName.txt" name
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: And the attachment is named "myFileName.txt"

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/DownloadServlet.java [hunk 3]
```diff
@@ -60,22 +60,23 @@
...
 
-    @VisibleForTesting String blobIdFrom(String pathInfo) {
-        return pathInfo.substring(1);
-    }
-
-    @VisibleForTesting void download(MailboxSession mailboxSession, String blobId, HttpServletResponse resp) {
+    @VisibleForTesting void download(MailboxSession mailboxSession, DownloadPath downloadPath, HttpServletResponse resp) {
+        String blobId = downloadPath.getBlobId();
         try {
+            addHeader(downloadPath.getName(), resp);
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/DownloadServlet.java]: it's not "a header" it's filename

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java [hunk 4]
```diff
@@ -82,6 +84,7 @@ private Attachment attachment(Row row) {
                 .attachmentId(AttachmentId.from(row.getString(ID)))
                 .bytes(row.getBytes(PAYLOAD).array())
                 .type(row.getString(TYPE))
+                .name(Optional.fromNullable(row.getString(NAME)))
                 .size(row.getLong(SIZE))
```
Comment (mbaechler) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java]: I'm not happy about that, what if we screw up and size becomes != then array.length ?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature (hunk 2 above)
Comment (aduprat) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: OK

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/DownloadServlet.java (hunk 3 above)
Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/DownloadServlet.java]: In the context of a http request (handled by a servlet), it's an header.

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java (hunk 4 above)
Comment (aduprat) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java]: OK

**Comment by aduprat** [comment]
#305

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/DownloadServlet.java (hunk 3 above)
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/DownloadServlet.java]: Ok, it's both a header and a filename, just choose the right method name then

Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/DownloadServlet.java]: OK

**Pull request closed without merging**